automation.md

```diff
@@ -0,0 +1,102 @@
+This is the architecture of Java Access Bridge from oracle official site:
+
+![jab architecture](img/jab-block-diagram.gif)
+
+To learn more about JAB, please visit [oracle](https://docs.oracle.com/javase/8/docs/technotes/guides/access/jab/introduction.html#jab-overview)
+
+# Web Automation
+Web automation mainly depends on the type of browser, currently there are two main categories:
+
+## Internet Explorer(IE)  
+
+Through javascript injection, the difficulty is that there are too many versions of IE, and it is still technically difficult to make multiple versions of IE compatible.
+
+## Chromium-based browsers
+
+There are generally three ways:
+- Selenium webdriver
+
+To learm more about webdriver, please visit [WebDriver](https://w3c.github.io/webdriver/), [Chrome webdriver](https://chromedriver.chromium.org/).
+
+- Chrome devtool protocol(CDP)
+
+Many tools use CDP, see [awesome-chrome-devtools](https://github.com/ChromeDevTools/awesome-chrome-devtools).
+Related to web automation testing are Puppeteer, Playwright.
+The following is the playwright python code to start the Chrome browser:
+```python
+browser = p.chromium.launch(channel="chrome", headless=False)
+context = browser.new_context()
+# Open new page
+page = context.new_page()
+# Go to https://www.google.com/
+page.goto("https://www.google.com/")
+```
+Using ProcessExplorer, you can see that the chrome start parameters are as follows,
+```
+"C:\Program Files\Google\Chrome\Application\chrome.exe" --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,AcceptCHFrame,AutoExpandDetailsElement --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --no-sandbox --user-data-dir=C:\Users\test\AppData\Local\Temp\playwright_chromiumdev_profile-V3LZUg **--remote-debugging-pipe** --no-startup-window
+```
+According to the chrome startup parameter description:
+```
+--remote-debugging-pipe: Enables remote debug over stdio pipes [in=3, out=4]. Optionally, specifies the format for the protocol messages, can be either "JSON" (the default) or "CBOR"
+```
+
+How to use CDP, you can get started from [this link](https://github.com/aslushnikov/getting-started-with-cdp/blob/master/README.md).
```

Comment:
For the way to use CDP, you can get started from [this link](https://github.com/aslushnikov/getting-started-with-cdp/blob/master/README.md)